### PR TITLE
Add Docker client library for odo

### PR DESCRIPTION
**Pull request by johnmcollier**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:

/kind feature

**What does does this PR do / why we need it**:
This PR creates a client library interface (called `lclient` currently, open to better names 😉 ) for odo to interact with Docker. This PR pulls in the official `github.com/docker/docker` library and dependencies and uses it to interact with containers

- Vendors `github.com/docker/docker` and its dependencies (first time doing this, let me know if I messed this part up). First two commits cover this.
    - Vendored via `glide get github.com/docker/docker --strip-vendor`
- Implements a basic `lclient` package in odo to interact with Docker
    - Functions to create container, retrieve container with a given label, and pull images. Will expand as needed
- Unit tests for the functions I added, as well as a stub mock Docker client
    - Mock client was adapted from how we handled it in the Codewind project (https://github.com/eclipse/codewind-installer/blob/master/pkg/docker/docker_test.go)
- No integration tests at the moment, as nothing is actually using this code currently

**Which issue(s) this PR fixes**:

Fixes https://github.com/openshift/odo/issues/2578

### CODE REVIEW

File: pkg/lclient/fakeclient.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
```
Comment (cdrage) [inline pkg/lclient/fakeclient.go]: Please confirm with legal that we can have this copyright here. I don't think it'd be compatible with our current license.

Comment (johnmcollier) [reply pkg/lclient/fakeclient.go]: That was added erroneously and should not be there, removed.

File: pkg/lclient/containers.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+import (
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/pkg/errors"
+)
+
+// GetContainersByComponent returns the list of Docker containers that matches the specified component label
+// If no container with that name exists, it returns an error
```
[the remark this answers is not in the document]
Comment (johnmcollier) [reply pkg/lclient/containers.go]: Yeah, I toyed with a couple approaches for this function but forgot to update the function description afterwards.

File: pkg/lclient/containers_test.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+					Labels: map[string]string{
+						"component": "node",
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:           "Case 2: Invalid component name",
+			client:         fakeClient,
```
[the remark this answers is not in the document]
Comment (johnmcollier) [reply pkg/lclient/containers_test.go]: As discussed on slack, split the function in two to make unit testing easier.

File: pkg/lclient/client.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+// New creates a new instances of Docker client, with the minimum API version set to
+//   to the value of MinDockerAPIVersion.
+func New() (*Client, error) {
+	// Create the context and client variables for docker
+	ctx := context.Background()
+
+	// Create a new Docker client instance
+	tempClient, err := client.NewClientWithOpts(client.WithVersion(MinDockerAPIVersion))
```
[the remark this answers is not in the document]
Comment (johnmcollier) [reply pkg/lclient/client.go]: Just bad naming on my part, how about just `client`?

File: pkg/lclient/images.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+// PullImage uses Docker to pull the specified image. If there are any issues pulling the image,
+// it returns an error.
+func (dc *Client) PullImage(image string) error {
+
+	codewindOut, err := dc.Client.ImagePull(dc.Context, image, types.ImagePullOptions{})
+
+	if err != nil {
+		return errors.Wrapf(err, "Unable to pull docker image")
```
[the remark this answers is not in the document]
Comment (johnmcollier) [reply pkg/lclient/images.go]: Yeah, good point.

File: pkg/lclient/images.go [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+	}
+
+	defer codewindOut.Close()
+	if glog.V(4) {
+		_, err := io.Copy(os.Stdout, codewindOut)
```
[the remark this answers is not in the document]
Comment (johnmcollier) [reply pkg/lclient/images.go]: Fixed 👍

File: pkg/lclient/images.go [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+// it returns an error.
+func (dc *Client) PullImage(image string) error {
+
+	codewindOut, err := dc.Client.ImagePull(dc.Context, image, types.ImagePullOptions{})
+
+	if err != nil {
+		return errors.Wrapf(err, "Unable to pull docker image")
+	}
+
+	defer codewindOut.Close()
```
[the remark this answers is not in the document]
Comment (johnmcollier) [reply pkg/lclient/images.go]: Fixed 👍

File: pkg/lclient/images.go [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/docker/docker/api/types"
+	"github.com/pkg/errors"
+	"k8s.io/klog/glog"
+)
+
+// PullImage uses Docker to pull the specified image. If there are any issues pulling the image,
+// it returns an error.
+func (dc *Client) PullImage(image string) error {
+
+	codewindOut, err := dc.Client.ImagePull(dc.Context, image, types.ImagePullOptions{})
```
[the remark this answers is not in the document]
Comment (johnmcollier) [reply pkg/lclient/images.go]: Gah, good catch, my bad for missing this. Fixed

File: pkg/lclient/client.go [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/api/types/registry"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+)
+
+const errorMsg = `
```
Comment (adisky) [inline pkg/lclient/client.go]: const?

**Pull request merged**